src/index.ts

```diff
@@ -37,34 +50,73 @@ export const runScenario = (
             );
             process.exit(1);
         }
-        getLogger('unknown').info(
-            `Starting scenario: ${scenarioPath} (actions: ${actionDir}, out: ${outDir}, envConfig: ${envConfigFile})`,
+
+        getLogger('setup').info(
+            `RUNNING: scenario(s): ${scenarioPaths} (actions: ${actionDir}, out: ${outDir}, envDir: ${envConfigDir}, numberOfScenariosRunInParallel: ${numberOfScenariosRunInParallel}, environmentNameToBeUsed: ${environmentNameToBeUsed})`,
         );
 
         OUT_DIR = outDir;
 
-        const envConfig = envConfigFile
-            ? loadYamlConfiguration(envConfigFile)
+        const envConfig = envConfigDir
+            ? loadYamlConfiguration(
+                  `${envConfigDir}/${environmentNameToBeUsed}.yaml`,
+              )
             : {};
+        getLogger('setup').debug(
+            `Using '${environmentNameToBeUsed}' configuration: ${stringify(
+                envConfig,
+            )}`,
+        );
 
         const actions: Action[] = loadAllActions(actionDir, envConfig);
+        getLogger('setup').debug(
+            `Successfully loaded ${actions.length} actions`,
+        );
 
-        const scenarios: Scenario[] = scenarioPath.endsWith('yaml')
-            ? loadScenariosById(scenarioPath, actions)
-            : loadAllScenarios(scenarioPath, actions);
+        scenarioPaths.forEach(scenarioPath => {
+            const scenario: Scenario[] = scenarioPath.endsWith('yaml')
+                ? loadScenariosById(scenarioPath, actions)
+                : loadAllScenarios(scenarioPath, actions);
+            getLogger('setup').debug(
+                `Successfully loaded ${
+                    scenario.length
+                } scenario(s): ${scenarioPath}`,
+            );
 
-        processScenarios(scenarios);
+            processScenarios(scenario, numberOfScenariosRunInParallel);
+        });
     } catch (e) {
-        getLogger('unknown').error(e);
+        getLogger('setup').error(e);
     }
 };
 
-async function processScenarios(scenarios: Scenario[]): Promise<void> {
-    for (let i = 0; i < scenarios.length; i += NUM_PARALLEL_RUNS) {
+export const runScenario = (
+    scenarioPath: string,
+    actionDir: string,
+    outDir = 'out',
+    envConfigFile: string,
+): void => {
+    runMultipleSceanriosWithConfig(
+        actionDir,
+        outDir,
+        envConfigFile.substring(
+            0,
+            envConfigFile.length - 12,
+        ) /* substracting '/config.yaml' from the string */,
```

Comment:
Instead of using a comment, extract to a function with a selfexplanatory name.

Reply:
method is deprecated and will be deleted soon